soc/xtensa/intel_adsp/tools/cavstool.py

```diff
@@ -18,11 +18,160 @@
 
 PAGESZ = 4096
 HUGEPAGESZ = 2 * 1024 * 1024
-HUGEPAGE_FILE = "/dev/hugepages/cavs-fw-dma.tmp"
+HUGEPAGE_FILE = "/dev/hugepages/cavs-fw-dma.tmp."
 
 # Log is in the fourth window, they appear in 128k regions starting at 512k
 WINSTREAM_OFFSET = (512 + (3 * 128)) * 1024
 
+class HDAStream:
```

Comment:
This looks like a perfect generalization of the stream from the limited output stream definition we had earlier.  But shouldn't we also be removing the older code from the loader and using this instead?

Also: this much surgery wants to be its own patch.  I'd suggest a refactoring that introduces this class and uses it instead of the older code, then dropping new protocols on top of it.

Reply:
To be frank, I was frightened by the time it might take to do just that because of the seemingly endless hardware quirks. But it definitely tries to abstract it, and yes I should refactor the existing firmware load into this to some degree.

Reply:
I'm happy to revisit this idea in a cleanup round, this patchset I feel like should focus on providing the basics of HDA at this point.